pkg/eval/mods/file/file_test.go

```diff
@@ -12,6 +12,12 @@ func TestFile(t *testing.T) {
 		ev.AddGlobal(eval.NsBuilder{}.AddNs("file", Ns).Ns())
 	}
 	evaltest.TestWithSetup(t, setup,
-		evaltest.That(`file:fopen`).Throws(evaltest.AnyError),
+		evaltest.That(`echo haha > out3`, `f = (file:open out3)`, `slurp <$f`, ` file:close $f`).Puts("haha\n"),
+		//No name is specified hence should give an error.
+		evaltest.That(`file:open`).Throws(evaltest.AnyError),
+		evaltest.That(`file:close`).Throws(evaltest.AnyError),
```

Comment:
That's good for your first attempt at writing an Elvish unit test but not ideal. It's better to verify the actual exception matches what we expect rather than matching any exception. In this case we want to match a `errs.ArityMismatch` error as shown in this example:

https://github.com/elves/elvish/blob/8a7073a20fb224f0a202f5c7aaf92bf7bc56d47d/pkg/eval/compile_effect_test.go#L146-L151

Although the "Actual" value should be zero for these two tests.

Reply:
Thank you sir. I am learning from you.